### PR TITLE
adding_torchaudio_pkg

### PR DESCRIPTION
**Pull request by ngam**
Checklist
- [ ] Title of this PR is meaningful: e.g. "Adding my_nifty_package", not "updated meta.yaml".
- [ ] License file is packaged (see [here](https://github.com/conda-forge/staged-recipes/blob/main/recipes/example/meta.yaml#L57-L66) for an example).
- [ ] Source is from official source.
- [ ] Package does not vendor other packages. (If a package uses the source of another package, they should be separate packages or the licenses of all packages need to be packaged).
- [ ] If static libraries are linked in, the license of the static library is packaged.
- [ ] Build number is 0.
- [ ] A tarball (`url`) rather than a repo (e.g. `git_url`) is used in your recipe (see [here](https://conda-forge.org/docs/maintainer/adding_pkgs.html#build-from-tarballs-not-repos) for more details).
- [ ] GitHub users listed in the maintainer section have posted a comment confirming they are willing to be listed there.
- [ ] When in trouble, please check our [knowledge base documentation](https://conda-forge.org/docs/maintainer/knowledge_base.html) before pinging a team.

### CODE REVIEW

File: recipes/torchaudio/meta.yaml [hunk 1]
```diff
@@ -0,0 +1,80 @@
...
+        - pytorch =*={{ torch_proc_type }}*      # [build_platform != target_platform]
+
+      host:
+        - python
+        - pip
+        - setuptools
+        - cmake
+        - ninja
+        - cudnn                                  # [cuda_compiler_version != "None"]
+        - defaults::numpy >=1.11
```
Comment (hmaarrfk) [inline recipes/torchaudio/meta.yaml:43]:
```suggestion
        - numpy
```

File: recipes/torchaudio/meta.yaml [hunk 2]
```diff
@@ -0,0 +1,80 @@
...
+        - pytorch
+        - pytorch-gpu                            # [cuda_compiler_version != "None"]
+        - pytorch-cpu                            # [cuda_compiler_version == "None"]
+      run:
+        - python
+        - defaults::numpy >=1.11
```
Comment (hmaarrfk) [inline recipes/torchaudio/meta.yaml:49]:
```suggestion
```

File: recipes/torchaudio/meta.yaml [hunk 3]
```diff
@@ -0,0 +1,80 @@
...
+
+source:
+  url: https://github.com/pytorch/audio/archive/v0.10.0.tar.gz
+  sha256: 9d0413434eefd2a483bf4006440c02940d15e597ab75703aa3ae2c6d5297e7ad
+
+build:
+  number: 0
+  skip: true  # [win]
+
+outputs:
```
Comment (hmaarrfk) [inline recipes/torchaudio/meta.yaml:17]: Why are you bothering with outputs? Typically they help fix mistakes or help split a build up into many. I don't think you need them here.

Comment (ngam) [reply recipes/torchaudio/meta.yaml:17]: > Why are you bothering with outputs? Typically they help fix mistakes or help split a build up into many. I don't think you need them here.

I am not really sure 😶‍🌫️ 

I am testing random ideas to see what sticks. For some reason, when I didn't have the outputs it kept failing. I will try without the outputs soon. I am also really bad at squashing commits, so this is a total mess (I am sorry, but thank you so much for helping!)

Comment (ngam) [reply recipes/torchaudio/meta.yaml:17]: Oh, now I remember: How do I call the scripts outside the requirements? Like torchvision, there are two scripts that should be called I assume. Or am I reading this wrong?

Comment (ngam) [reply recipes/torchaudio/meta.yaml:17]: Never mind, I think I can do it like in line 35 of the example

Comment (hmaarrfk) [reply recipes/torchaudio/meta.yaml]: I think you just call them `build.sh` and `bld.bat`

See a simple recipe:
https://github.com/conda-forge/check-feedstock/blob/master/recipe/meta.yaml#L19

File: recipes/torchaudio/build-torch.bat [hunk 4]
```diff
@@ -0,0 +1,2 @@
+set "TORCHVISION_INCLUDE=%LIBRARY_INC%"
```
Comment (hmaarrfk) [inline recipes/torchaudio/build-torch.bat:1]: Rename this flie `bld.bat`

File: recipes/torchaudio/build-torch.sh [hunk 5]
```diff
@@ -0,0 +1,25 @@
+if [[ "$cuda_compiler_version" == "None" ]]; then
```
Comment (hmaarrfk) [inline recipes/torchaudio/build-torch.sh:1]: Rename this file `build.sh`

File: recipes/torchaudio/meta.yaml [hunk 6]
```diff
@@ -0,0 +1,75 @@
...
+  version: {{ version }}
+
+source:
+  url: https://github.com/pytorch/audio/archive/v0.10.0.tar.gz
+  sha256: 9d0413434eefd2a483bf4006440c02940d15e597ab75703aa3ae2c6d5297e7ad
+
+build:
+  number: 0
+  script: build-torch.sh   # [not win]
+  script: build-torch.bat  # [win]
```
Comment (hmaarrfk) [inline recipes/torchaudio/meta.yaml:16]:
```suggestion
  script: build.sh   # [not win]
  script: bld.bat    # [win]
```

File: recipes/torchaudio/build-torch.sh (hunk 5 above)
Comment (ngam) [reply recipes/torchaudio/build-torch.sh]: ✅

File: recipes/torchaudio/build-torch.bat (hunk 4 above)
Comment (ngam) [reply recipes/torchaudio/build-torch.bat]: ✅

**Comment by ngam** [comment]
@hmaarrfk
I get this error now. I think pytorch-mutex is needed for another package/function. Let me know if you have any further ideas. I will be stepping away for now, and I will resume trying in a few hours or tomorrow. Thanks a lot!

```
with channels:
 - local
 - conda-forge
 - conda-forge

The reported errors are:
   Encountered problems while solving:
     - nothing provides requested pytorch-mutex 1.0.* cpu
```

**Comment by ngam** [comment]
> @hmaarrfk I get this error now. I think pytorch-mutex is needed for another package/function. Let me know if you have any further ideas. I will be stepping away for now, and I will resume trying in a few hours or tomorrow. Thanks a lot!
> 
> ```
> with channels:
>  - local
>  - conda-forge
>  - conda-forge
> 
> The reported errors are:
>    Encountered problems while solving:
>      - nothing provides requested pytorch-mutex 1.0.* cpu
> ```

Does this mean one would need to make a conda-forge for pytorch-mutex too??

**Comment by hmaarrfk** [comment]
mutexes are typically used for packages with different names.

pytorch, acts as a natural mutex.

**Pull request closed without merging**